packages/utils/src/bit_shifts.cairo

```diff
@@ -113,9 +131,89 @@ pub fn fast_pow<
     }
 }
 
+
+/// Fast power of 2 using lookup tables
+/// Reference: https://github.com/keep-starknet-strange/alexandria/pull/336
+///
+/// # Arguments
+/// * `exponent` - The exponent to raise 2 to
+/// # Returns
+/// * `u64` - The result of 2^exponent
+/// # Panics
+/// * If `exponent` is greater than 63 (out of the supported range)
+pub fn pow2(exponent: u32) -> u64 {
+    let hardcoded_results: [u64; 64] = [
```

Comment:
Why this is not a const?

Reply:
I thought it would be cleaner to keep it with the function as we shouldn't use it outside, but I can move it out if you want. step wise it's the same, I guess the compiler simplifies down to the same thing.

Reply:
Even if you call pow2 multiple times?

Reply:
> Even if you call pow2 multiple times?

Correct, I tried with `test_pow2` which performs seven exponentiations. I get the exact same steps number.

Reply:
Please have a look here: https://github.com/feltroidprime/cairo-perfs-snippets/blob/main/src/pow_2_lookup.cairo. It seems like using match instead of an lookup might be better.

Reply:
Thanks for the link, very informative. I think for my use case might be better, however for raito it still makes sense to use a lookup table because you are going to use pow2 many times across execution (validating the whole chain).

If you access the function more than twice, it becomes cheaper, each Additional access costs 1 RC + 7 steps, which is cheaper than 11 steps + 1 RC for the match.